apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.js

```diff
@@ -1,30 +1,44 @@
 // @ts-check
-const ERR = require('async-stacktrace');
 const asyncHandler = require('express-async-handler');
 import * as express from 'express';
-import * as async from 'async';
 import * as error from '@prairielearn/error';
 import { startTestQuestion } from '../../lib/question-testing';
 import * as sqldb from '@prairielearn/postgres';
 import * as path from 'path';
-const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'));
-import { logger } from '@prairielearn/logger';
+import { z } from 'zod';
+
 import { QuestionRenameEditor, QuestionDeleteEditor, QuestionCopyEditor } from '../../lib/editors';
 import { config } from '../../lib/config';
 import { encodePath } from '../../lib/uri-util';
 import { idsEqual } from '../../lib/id';
 import { generateSignedToken } from '@prairielearn/signed-token';
 import { copyQuestionBetweenCourses } from '../../lib/copy-question';
-import { callbackify } from 'node:util';
 import { flash } from '@prairielearn/flash';
 import { features } from '../../lib/features/index';
 import { getCanonicalHost } from '../../lib/url';
 import { isEnterprise } from '../../lib/license';
 import { selectCoursesWithEditAccess } from '../../models/course';
+import { IdSchema } from '../../lib/db-types';
 
 const router = express.Router();
 const sql = sqldb.loadSqlEquiv(__filename);
 
+const SelectedAssessmentsSchema = z
+  .array(
+    z.object({
+      title: z.string(),
+      course_instance_id: IdSchema,
+      assessments: z.array(
+        z.object({
+          assessment_id: IdSchema,
+          color: z.string(),
+          label: z.string(),
+        }),
+      ),
+    }),
+  )
+  .nullable();
```

Comment:
It's a bit weird to have a query type use `.array()` at the top level; typically we'd want the schema to correspond to one row in the output. However, in this particular case, it looks like the query is actually pre-aggregating the results. A suggestion to make this less bespoke: drop the `jsonb_agg` from the `select_assessments_with_question_for_display` query and just directly return normal rows. You'll need to change `queryRow` to `queryRows` and then change the schema here.

Reply:
Updated the query to drop the array and updated `queryRow` to `queryRows` in [15e6a55](https://github.com/PrairieLearn/PrairieLearn/pull/9439/commits/15e6a551ace1b217ed76fec735c5306f682080d1).